src/services/agent-manager/index.ts

```diff
@@ -49,14 +51,41 @@ export interface AgentManagerItems {
  * @example
  * const agentManager = await createAgentManager('id-agent123', { auth: { type: 'key', clientKey: '123', externalId: '123' } });
  */
-export async function createAgentManager(agent: string, options: AgentManagerOptions): Promise<AgentManager> {
+export async function createAgentManager(agent: string, managerOptions: AgentManagerOptions): Promise<AgentManager> {
     let firstConnection = true;
 
+    let chatRequestPending = false;
+    let queuedInterrupt = false;
+    let currentVideoId: string | undefined;
+
+    const resetInterruptState = (): void => {
+        chatRequestPending = false;
+        queuedInterrupt = false;
+        currentVideoId = undefined;
+    };
+
+    const options: AgentManagerOptions = {
```

Comment:
I think we have another convention for doing those sort of stuff - i.e. getting in the middle of a callback call.
I can't remember where but I remember we do have one. Can you check if I'm right? If so - update it.

Reply:
which new version?